csr/csr.go

```diff
@@ -114,19 +114,18 @@ func NamesFromCSR(csr *x509.CertificateRequest) names {
 	if csr.Subject.CommonName != "" {
 		sans = append(sans, csr.Subject.CommonName)
 	}
-	sans = core.UniqueLowerNames(sans)
 
 	if csr.Subject.CommonName != "" {
-		return names{SANs: sans, CN: strings.ToLower(csr.Subject.CommonName)}
+		return names{SANs: core.UniqueLowerNames(sans), CN: strings.ToLower(csr.Subject.CommonName)}
 	}
 
 	// If there's no CN already, but we want to set one, promote the first SAN
 	// which is shorter than the the maximum acceptable CN length (if any).
 	for _, name := range sans {
 		if len(name) <= maxCNLength {
-			return names{SANs: sans, CN: name}
+			return names{SANs: core.UniqueLowerNames(sans), CN: name}
```

Comment:
Can you elaborate a little more in the comment starting at line 122 please? Due to line 117 being removed, we now no longer promote the alphabetically-first SAN to be the CN, but instead promote the first SAN that matches the length check to be the CN.